app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -2882,12 +2946,33 @@ public void onScrolledDown(final RecyclerView recyclerView) {
                 if (playQueue != null && !playQueue.isComplete()) {
                     playQueue.fetch();
                 } else if (binding != null) {
-                    binding.playQueue.clearOnScrollListeners();
+                    binding.itemsList.clearOnScrollListeners();
                 }
             }
         };
     }
 
+    private StreamSegmentAdapter.StreamSegmentListener getStreamSegmentListener() {
+        return (item, seconds) -> {
+            segmentAdapter.selectSegment(item);
+            seekTo(seconds * 1000);
+            triggerProgressUpdate();
+        };
+    }
+
+    private int getNearestStreamSegmentPosition(final long playbackPosition) {
+        int nearestPosition = 0;
+        final List<StreamSegment> segments = currentMetadata.getMetadata().getStreamSegments();
+
+        for (int i = 0; i < segments.size(); i++) {
+            if (segments.get(i).getStartTimeSeconds() * 1000 > playbackPosition) {
+                break;
+            }
+            nearestPosition++;
+        }
+        return Math.max(0, nearestPosition - 1);
```

Comment:
Just to make sure: is it possible that the first segment does not start at 0:00? In that case the result of this function would be wrong when the progress is 0:00, since the first segment has yet to start. Maybe no item should be selected in the adapter in that case?

Reply:
I ran multiple checks myself when I added the functionality to the extractor by uploading a video and playing around. The chapters are only available if the description contains 0:00, so it's always starting from the beginning.

Reply:
Ok, great, then this is good